core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -436,19 +491,53 @@ public UfsManager.UfsClient getUfsClient(long mountId) {
   /**
    * Resolves the given Alluxio path. If the given Alluxio path is nested under a mount point, the
    * resolution maps the Alluxio path to the corresponding UFS path. Otherwise, the resolution is a
-   * no-op.
+   * no-op. It will call {@link MountTable#resolve(AlluxioURI, List)}.
    *
-   * @param uri an Alluxio path URI
+   * @param alluxioLockedInodePath an Alluxio LockedInodePath
+   * @return the {@link Resolution} representing the UFS path
+   * @throws InvalidPathException if an invalid path is encountered
+   */
+  public Resolution resolve(LockedInodePath alluxioLockedInodePath) throws InvalidPathException {
+    if (!alluxioLockedInodePath.fullPathExists()) {
+      return resolve(alluxioLockedInodePath.getUri(), Collections.emptyList());
+    }
+    return resolve(alluxioLockedInodePath.getUri(), alluxioLockedInodePath.getInodeViewList());
+  }
+
+  /**
+   * Resolves the given Alluxio path. If the given Alluxio path is nested under a mount point, the
+   * resolution maps the Alluxio path to the corresponding UFS path. Otherwise, the resolution is a
+   * no-op. It will call {@link MountTable#resolve(AlluxioURI, List)}.
+   *
+   * @param uri an Alluxio LockedInodePath
    * @return the {@link Resolution} representing the UFS path
    * @throws InvalidPathException if an invalid path is encountered
    */
   public Resolution resolve(AlluxioURI uri) throws InvalidPathException {
+    return resolve(uri, Collections.emptyList());
+  }
+
+  /**
+   * Underlying implementation of resolve. If the calling context has no LockedInodePath, then
+   * call this method and pass an empty array list as the second parameter.
+   * @param uri target alluxio path's uri
+   * @param inodeViewList target alluxio path's inodes
+   * @return the {@link Resolution} representing the UFS path
+   * @throws InvalidPathException if an invalid path is encountered
+   */
```

Comment:
if this is private, no need for javadoc unless there's really something tricky in the code